src/pages/dataTableUtilities/actions/cellActions.js

```diff
@@ -129,6 +131,102 @@ export const editTransactionBatchLocation = (location, route) =>
 export const editStocktakeBatchLocation = (location, route) =>
   editBatchLocation(location, 'StocktakeBatch', route);
 
+export const editOpeningStock = (value, rowKey, route) => (dispatch, getState) => {
+  const { data, keyExtractor } = selectPageState(getState());
+
+  const requisitionItem = data.find(row => keyExtractor(row) === rowKey);
+  if (requisitionItem) {
+    UIDatabase.write(() => {
+      const didUpdate = requisitionItem.setOpeningStock(parsePositiveInteger(value));
+      UIDatabase.save('RequisitionItem', requisitionItem);
```

Comment:
just a query: is it ok to save to database when the user enters invalid values?

Reply:
`UIDatabase.save()` is a little misleading. It's purpose is to *only* trigger events, it doesn't actually save anything. The values are actually being set to `0` in `RequisitionItem`.

I'm thinking that this might be less ideal and actually saving the 'invalid values' and disallowing finalizing if invalid values are present (Where closing stock < 0) would be better - entering values for `opening`, `incoming`, `outgoing` and then `positive adjustment` and then entering a number just `1` too many in the `negative adjustment` column, resetting all the columns would be incredibly frustrating..!